src/build.ts

```diff
@@ -53,7 +53,13 @@ export async function copyPublicAssets(nitro: Nitro) {
         cwd: srcDir,
         absolute: false,
         dot: true,
-        ignore: nitro.options.ignore,
+        ignore: nitro.options.ignore
+          .map((p) =>
+            p.startsWith("*") || p.startsWith("!*")
+              ? p
+              : relative(srcDir, resolve(nitro.options.srcDir, p))
+          )
+          .filter((p) => !p.startsWith("../")),
```

Comment:
This might make issues with multi-layer when relative path is outside of main `srcDir` but inside other layer's source dir... But let's try and see!